pkg/networking/middleware/retry_middleware.go

```diff
@@ -0,0 +1,108 @@
+package middleware
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/cenkalti/backoff/v5"
+
+	"github.com/snyk/go-application-framework/pkg/configuration"
+)
+
+const defaultRetryCount uint = 3
+const configurationKeyRetryCount = "INTERNAL_MAX_NET_RETRIES"
+const defaultRetryAfterSeconds = 5
+
+type RetryMiddleware struct {
+	next   http.RoundTripper
```

Comment:
nitpick: I'd rather not use `next`, but instead something like roundTripper. Next may be usual in many http filter chains, I still find it unintuitive.